databricks/koalas/indexes.py

```diff
@@ -641,6 +641,71 @@ def copy(self, name=None):
             result.name = name
         return result
 
+    def droplevel(self, level):
+        """
+        Return index with requested level(s) removed.
+        If resulting index has only 1 level left, the result will be
+        of Index type, not MultiIndex.
+
+        Parameters
+        ----------
+        level : int, str, or list-like, default 0
+            If a string is given, must be the name of a level
+            If list-like, elements must be names or indexes of levels.
+
+        Returns
+        -------
+        Index or MultiIndex
+
+
+        Examples
+        --------
+        >>> midx = ks.DataFrame({'a': ['a', 'b']}, index=[['a', 'x'], ['b', 'y'], [1, 2]]).index
+        >>> midx  # doctest: +SKIP
+        MultiIndex([('a', 'b', 1),
+                    ('x', 'y', 2),
+                   )
+
+        >>> midx.droplevel([0, 1])  # doctest: +SKIP
+        Int64Index([1, 2], dtype='int64')
+
+        >>> midx.droplevel(0)  # doctest: +SKIP
+        MultiIndex([('b', 1),
+                    ('y', 2)],
+                   )
+        """
+        names = self.names
+        nlevels = self.nlevels
+        if not isinstance(level, (tuple, list)):
+            level = [level]
+
+        for n in level:
+            if isinstance(n, int) and (n > nlevels - 1):
+                raise IndexError("Too many levels: Index has only {} levels, not {}"
+                                 .format(nlevels, n+1))
+            if isinstance(n, str) and (n not in names):
+                raise KeyError("Level {} not found".format(n))
+
+        if len(level) >= nlevels:
+            raise ValueError(
+                "Cannot remove {} levels from an index with {} "
+                "levels: at least one level must be "
+                "left.".format(len(level), nlevels)
+            )
+        int_level = [n if isinstance(n, int) else names.index(n) for n in level]
+        index_columns = self._internal.index_columns.copy()
+        index_columns = [index_columns[c] for c in range(0, nlevels)
+                         if c not in int_level]
```

Comment:
We should use `self._internal.index_scols` instead of `self._internal.index_columns`.